lib/note-content-editor.jsx

```diff
@@ -0,0 +1,68 @@
+import React, { PropTypes } from 'react';
+import { Editor, EditorState, ContentState } from 'draft-js';
+import { urlDecorator } from './url-decorator';
+
+function plainTextContent( editorState ) {
+	return editorState.getCurrentContent().getPlainText( '\n' )
+}
+
+export default React.createClass( {
+
+	propTypes: {
+		content: PropTypes.string.isRequired,
+		onChangeContent: PropTypes.func.isRequired
+	},
+
+	getInitialState() {
+		const { content } = this.props;
+		const contentState = ContentState.createFromText( content, '\n' );
+		return {
+			editorState: EditorState.createWithContent( contentState, urlDecorator )
+		}
+	},
+
+	handleEditorStateChange( editorState ) {
+		const contentHasChanged = plainTextContent( editorState )
+			!== plainTextContent( this.state.editorState )
+
+		this.setState( { editorState }, () => {
+			if ( contentHasChanged ) {
+				this.props.onChangeContent( plainTextContent( editorState ) );
+			}
+		} );
+	},
+
+	componentWillReceiveProps( nextProps ) {
+		if (
+			// if we've received this content before there's nothing to do now
+			nextProps.content !== this.props.content
+			// if the new content isn't different from what's rendered
+			// don't change it because we'd lose the SelectionState
+			&& plainTextContent( this.state.editorState ) !== nextProps.content
+		) {
+			const contentState = ContentState.createFromText( nextProps.content, '\n' );
+			this.setState( {
+				editorState: EditorState.createWithContent( contentState, urlDecorator )
+			} );
+		}
+	},
+
+	focus() {
+		if ( !this.state.editorState.getSelection().getHasFocus() ) {
+			this.refs.editor.focus();
+		}
+	},
+
+	render() {
+		return (
+			<Editor
+				ref='editor'
```

Comment:
Let's not introduce new code with this style of `ref` attribute. Instead, let's use the function syntax. The React project strongly suggests this as well (for reasons which can be enumerated)

```js
import { invoke } from 'lodash';

saveEditorRef( ref ) {
	this.editor = ref;
}

focus() {
	invoke( this, 'editor' );
}

<Editor ref={ this.saveEditorRef } … />
```

Notice that we broke a convention of working through `this.setState()` which in this case is appropriate due to the stateful nature of the DOM elements with respect to the React component's _backing instance_.